tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,381 @@
+    type(product).model = PropertyMock(return_value="gateBox")
+    return (feature, "cover.gatebox_position")
+
+
+@pytest.fixture(name="gatecontroller")
+def gate_fixture():
+    """Return a gateController fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateController-2bee34e750b8-position",
+        full_name="gateController-position",
+        device_class="gate",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gate controller")
+    type(product).model = PropertyMock(return_value="gateController")
+    return (feature, "cover.gatecontroller_position")
+
+
+async def test_init_gatecontroller(gatecontroller, hass, config):
+    """Test gateController default state."""
+
+    _, entity_id = gatecontroller
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-gateController-2bee34e750b8-position"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "gateController-position"
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_GATE
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_OPEN
+    assert supported_features & SUPPORT_CLOSE
+    assert supported_features & SUPPORT_STOP
+
+    assert supported_features & SUPPORT_SET_POSITION
+    assert ATTR_CURRENT_POSITION not in state.attributes
+    assert state.state == STATE_UNKNOWN
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My gate controller"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "gateController"
+    assert device.sw_version == "1.23"
+
+
+async def test_init_shutterbox(shutterbox, hass, config):
+    """Test gateBox default state."""
+
+    _, entity_id = shutterbox
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-shutterBox-2bee34e750b8-position"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "shutterBox-position"
+    assert entry.device_class == DEVICE_CLASS_SHUTTER
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_OPEN
+    assert supported_features & SUPPORT_CLOSE
+    assert supported_features & SUPPORT_STOP
+
+    assert supported_features & SUPPORT_SET_POSITION
+    assert ATTR_CURRENT_POSITION not in state.attributes
+    assert state.state == STATE_UNKNOWN
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My shutter"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)